src/CMakeLists.txt

```diff
@@ -363,6 +363,18 @@ ADD_SUBDIRECTORY(irc)
 
 ADD_DEFINITIONS(-DAPP_VERSION="${APP_VERSION}" -DAPP_BUILD="${APP_BUILD}" -DAPP_TARGET="${APP_TARGET}")
 
+IF(UNIX)
+    SET(DATAROOTDIR "${CMAKE_INSTALL_PREFIX}/share" CACHE STRING "The platform-specific directory in which read-only data is generally installed")
+    SET(DATADIR "${DATAROOTDIR}/mudlet" CACHE STRING "Directory in which mudlet installs its read-only data")
+    SET(LUA_DEFAULT_DIR "${DATADIR}/lua" CACHE STRING "Directory in which mudlet installs its read-only lua scripts")
+ENDIF(UNIX)
+
+# Define a preprocessor symbol with the default fallback location from which
+# to load installed mudlet lua files. Set LUA_DEFAULT_DIR to a
+# platform-specific value. If LUA_DEFAULT_DIR is unset, the root directory
+# will be used.
+ADD_DEFINITIONS(-DLUA_DEFAULT_PATH="${LUA_DEFAULT_DIR}")
```

Comment:
Ideally we also want to provide the input for the "sudo make install" stuff that CMake will brew up given the right conditions - because we are trying to ensure that the *compiled in* value matches what the user actually *ends up with* - I'm not sure if I saved some low-level stuff to handle creating and copying the files from the src directory area into the specified destinations for AT LEAST the \*nix case (and ideally the MacOs one as well if one can work out the resource/package or whatever black magic THAT uses for *application bundles* :confounded: ).  The complication is that lua files are not compiled so the transformation from "source" to "destination" need a tiny bit more reading of the Cmake manual to work out how to do it, I'm sure it can be done but I for one would certainly need to do more reading to work that out...